### PR TITLE
Update uuid

### LINKED ISSUE #1623: Dependency conflict with uuid version 4.0.0
### Platform

Dart

### Obfuscation

Enabled

### Debug Info

Enabled

### Doctor

[√] Flutter (Channel stable, 3.10.2, on Microsoft Windows [Version 10.0.22621.2134], locale en-US)
    • Flutter version 3.10.2 on channel stable at C:\Users\User\flutter_windows_3.10.2-stable\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9cd3d0d9ff (3 months ago), 2023-05-23 20:57:28 -0700
    • Engine revision 90fa3ae28f
    • Dart version 3.0.2
    • DevTools version 2.23.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\User\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.6.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Pixel 4 (mobile)  • 9B191FFAZ0079Z • android-arm64  • Android 13 (API 33)
    • Windows (desktop) • windows        • windows-x64    • Microsoft Windows [Version 10.0.22621.2134]
    • Chrome (web)      • chrome         • web-javascript • Google Chrome 116.0.5845.141
    • Edge (web)        • edge           • web-javascript • Microsoft Edge 114.0.1823.51

[√] Network resources
    • All expected network resources are available.

• No issues found!

### Version

7.9.0

### Steps to Reproduce

Upgrade uuid to version 4.0.0 and run flutter pub get

### Expected Result

Dependencies to be resolved

### Actual Result

Resolving dependencies...
Because sentry_flutter >=7.9.0 depends on sentry 7.9.0 which depends on uuid ^3.0.0, sentry_flutter >=7.9.0 requires uuid ^3.0.0.
So, because flutter_test_app depends on both uuid ^4.0.0 and sentry_flutter ^7.9.0, version solving failed.

### Are you willing to submit a PR?

None

### PR DESCRIPTION
**Pull request by ueman**
## :bulb: Motivation and Context
Fixes #1623

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Unreleased
+
+- Enable compatibility with uuid v4 ([#xxx](https://github.com/getsentry/sentry-dart/pull/xxx))
```
Comment (ueman) [inline CHANGELOG.md]:
```suggestion
- Enable compatibility with uuid v4 ([#1647](https://github.com/getsentry/sentry-dart/pull/1647))
```

File: dart/pubspec.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ dependencies:
   http: '>=0.13.0 <2.0.0'
   meta: ^1.3.0
   stack_trace: ^1.10.0
-  uuid: ^3.0.0
+  uuid: '>=3.0.0 <5.0.0'
```
Comment (ueman) [inline dart/pubspec.yaml:18]: As far as I can tell, Sentry is not using any of the breaking changes, so we can keep supporting v3 while simultaneously supporting v4.

**Review by buenaflor — APPROVED** [review]
thx! looks good

**Comment by ueman** [comment]
Can you please merge it, since I don't have permission to do so?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Unreleased
+
```
Comment (vaind) [inline CHANGELOG.md:4]:
```suggestion

### Dependencies
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
...
+### Dependencies
```
Comment (vaind) [inline CHANGELOG.md:5]:
```suggestion
### Dependencies

```

**Comment by ueman** [comment]
Thanks @vaind 👌

**Pull request merged**